lib/commands/new.js

```diff
@@ -29,6 +30,27 @@ module.exports = Command.extend({
     '<app-name>'
   ],
 
+  beforeRun: function(rawArgs) {
+    if (!rawArgs.length) {
+      return;
+    }
+    // merge in blueprint availableOptions
+    var blueprint;
+    try{
+      blueprint = this.lookupBlueprint(rawArgs[0]);
+      this.registerOptions(blueprint);
+    }
+    catch(e) {
+      SilentError.debugOrThrow('ember-cli/commands/init', e);
```

Comment:
should likely be `ember-cli/commands/new`
both this and the `init` variant look very similar, maybe we should create a helper method that does this?